experimental/traffic-portal/src/app/core/topologies/topology-details/topology-details.component.spec.ts

```diff
@@ -0,0 +1,55 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { MatDialogModule } from "@angular/material/dialog";
+import { ActivatedRoute } from "@angular/router";
+import { RouterTestingModule } from "@angular/router/testing";
+import { ReplaySubject } from "rxjs";
+
+import { APITestingModule } from "src/app/api/testing";
+import {
+	NavigationService
+} from "src/app/shared/navigation/navigation.service";
+
+import { TopologyDetailsComponent } from "./topology-details.component";
+
+describe("TopologyDetailsComponent", () => {
+	let component: TopologyDetailsComponent;
+	let fixture: ComponentFixture<TopologyDetailsComponent>;
+	let route: ActivatedRoute;
+	let paramMap: jasmine.Spy;
```

Comment:
`route` and `paramMap` are never used in their declared scope (which could easily change when a non-trivial test is added).

Reply:
Using `paramMap` (and `route`) in test *existing topology*.